### PR TITLE
feat(podman): creating getVMType method

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

We have more and more provider specific implementation details, we were already using some kind of mechanism to detect if a machien was QEMU, this PR extend this mechanism to get the VMType from the `podman machine list`.

### What issues does this PR fix or reference?

Would be necessary for the future of https://github.com/containers/podman-desktop-extension-ai-lab/pull/1331#pullrequestreview-2160317866

Required for a better integration of https://github.com/containers/podman-desktop-extension-ai-lab/issues/1279

Fixes https://github.com/containers/podman-desktop-extension-ai-lab/issues/1338

### How to test this PR?

- [x] unit tests has been provided

### CODE REVIEW

File: packages/backend/src/managers/podmanConnection.ts [hunk 1]
```diff
@@ -98,6 +101,45 @@ export class PodmanConnection implements Disposable {
...
 
+  /**
+   * Get the VMType of the podman machine
+   * @param name the machine name, from {@link ContainerProviderConnection}
+   */
+  async getVMType(name?: string): Promise<VMType> {
+    const { stdout } = await process.exec(getPodmanCli(), ['machine', 'list', '--format', 'json']);
+
+    const parsed: unknown = JSON.parse(stdout);
+    if (!Array.isArray(parsed)) return VMType.UNKNOWN;
```
[the remark this answers is not in the document]
Comment (axel7083) [reply packages/backend/src/managers/podmanConnection.ts]: Yeah, I changed

**Review by lstocchi — APPROVED** [review]
LGTM

File: packages/shared/src/models/IPodman.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+export enum VMType {
+  WSL = 'wsl',
+  LIBKRUN = 'libkrun',
+  QEMU = 'qemu',
+  APPLEHV = 'applehv',
+  HYPERV = 'hyperv ',
```
Comment (jeffmaury) [inline packages/shared/src/models/IPodman.ts:24]: Is the extra space expected ?

Comment (axel7083) [reply packages/shared/src/models/IPodman.ts:24]: No, I will open a follow up PR to fix this, nice catch !

Comment (axel7083) [reply packages/shared/src/models/IPodman.ts:24]: See https://github.com/containers/podman-desktop-extension-ai-lab/pull/1346

**Pull request merged**